hydrus/app.py

```diff
@@ -467,24 +546,176 @@ def delete(self, path: str) -> Response:
                 except (ClassNotFound, InstanceNotFound) as e:
                     status_code, message = e.get_HTTP()
-                    return set_response_headers(jsonify(message), status_code=status_code)
+                    return set_response_headers(
+                        jsonify(message), status_code=status_code)
         abort(endpoint_['status'])
 
 
+class Items(Resource):
+
+    def put(self, path, int_list="") -> Response:
+        """
+        To insert multiple objects into the database
+        :param path: endpoint
+        :param int_list: Optional String containing ',' separated ID's
+        :return:
+        """
+        auth_response = check_authentication_response()
+        if isinstance(auth_response, Response):
+            return auth_response
+
+        endpoint_ = checkEndpoint("PUT", path)
+        if endpoint_['method']:
+            # If endpoint and PUT method is supported in the API
+            object_ = json.loads(request.data.decode('utf-8'))
+            object_ = object_["data"]
+            if path in get_doc().collections:
+                # If collection name in document's collections
+                collection = get_doc().collections[path]["collection"]
+                # title of HydraClass object corresponding to collection
+                obj_type = collection.class_.title
+                if validObjectList(object_):
+                    type_result = type_match(object_, obj_type)
+                    # If Item in request's JSON is a valid object
+                    # ie. @type is one of the keys in object_
+                    if type_result:
+                        # If the right Item type is being added to the
+                        # collection
+                        try:
+                            # Insert object and return location in Header
+                            object_id = crud.insert_multiple(
+                                objects_=object_, session=get_session(), id_=int_list)
+                            headers_ = [{"Location": get_hydrus_server_url(
+                            ) + get_api_name() + "/" + path + "/" + str(object_id)}]
+                            response = {
+                                "message": "Object with ID %s successfully added" %
+                                (object_id)}
+                            return set_response_headers(
+                                jsonify(response), headers=headers_, status_code=201)
+                        except (ClassNotFound, InstanceExists, PropertyNotFound) as e:
+                            status_code, message = e.get_HTTP()
+                            return set_response_headers(
+                                jsonify(message), status_code=status_code)
+
+                return set_response_headers(
+                    jsonify({400: "Data is not valid"}), status_code=400)
+
+            elif path in get_doc().parsed_classes and path + "Collection" not in get_doc().collections:

```

Comment:
If it's not a collection endpoint then the user shouldn't be able to insert multiple items. This should return this error instead.

Reply:
yes you are right , i missed that